### PR TITLE
Fixed autosuggest input with special characters

### LINKED ISSUE #1416: Autosuggest option values not working properly
**Describe the bug**
Writing certain strings in the autosuggest will show error because we are converting the value into a regular expression inside the component.

If we write for example `*`, the value of the autosuggest will be converted to regex. Because `*` is a wrong regular expression it will show this error.

**To Reproduce**
1. Create a new React App using `create-react-app`.
2. Install the `@dxc-technology\halstack-react` dependencies and run the app.
3. Write the following example:
```
  const countries = ["*", "Albania"];
  const [value, setValue] = useState("");
  const onChange = ({ value }) => {
    setValue(value);
  };
  const onBlur = ({ value }) => {
    setValue(value);
  };

  const callbackFunc = (newValue) => {
    const result = new Promise((resolve) =>
      setTimeout(() => {
        resolve(
          newValue
            ? countries.filter((option) =>
                option.toUpperCase().includes(newValue.toUpperCase())
              )
            : countries
        );
      }, 1500)
    );
    return result;
  };
  return (
    <DxcInset space="2rem">
      <DxcTextInput
        label="Enter your country"
        value={value}
        onChange={onChange}
        suggestions={callbackFunc}
        onBlur={onBlur}
      />
    </DxcInset>
  );
```
4. Click the component and select the option `*`.
5. See the error.

**Expected behaviour**
It should select the option.

**Screenshots**
image

image

image

image

**Add labels**
Text input, autosuggest.

### PR DESCRIPTION
**Pull request by Lisa159**
**Checklist**
_(Check off all the items before submitting)_
- [x] Build process is done without errors and all tests pass in `/lib` directory.
- [x] Self-reviewed the code prior to submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**
Special characters were transformed to regular expression so it was giving an error if they were invalid. 
Added `\` to escape the special characters. For example, if we write `*` in the input, internally the component will transform the value received to `\*` so it will be valid.
Added new test.

About the implementation, I have created a regular expression with the special characters that are giving error. Then I have converted it to string and created an array with each character. It is needed to remove the duplicate characters from the array because the function `.toString()` is literally converting the regular expression in a string without removing the characters `/`, `[` and `]` that are used to create de regular expresion (example, `const a = /[a-z]/`).

**Screenshots**
Created a new function to transform the value:
image

**Closes #1416**

### CODE REVIEW

**Comment by GomezIvann** [comment]
This is great stuff, Aida! 🚀 🚀 🚀

File: lib/src/text-input/TextInput.test.js [hunk 1]
```diff
@@ -643,6 +645,54 @@ describe("TextInput component synchronous autosuggest tests", () => {
     expect(input.value).toBe("");
     expect(queryByRole("listbox")).toBeFalsy();
   });
+
+  test("Autosuggest escapes special characters", () => {
+    const onChange = jest.fn();
+    const { getAllByText, getByText, getByRole } = render(
+      <DxcTextInput label="Autocomplete Countries" suggestions={special_characters} onChange={onChange} />
```
Comment (Jialecl) [inline lib/src/text-input/TextInput.test.js:652]: Change "special_characters" to camel case.

File: lib/src/text-input/TextInput.test.js [hunk 2]
```diff
@@ -643,6 +645,54 @@ describe("TextInput component synchronous autosuggest tests", () => {
...
+    );
+    const input = getByRole("combobox");
+    fireEvent.focus(input);
+    const list = getByRole("listbox");
+    fireEvent.change(input, { target: { value: "/" } });
+    expect(list).toBeTruthy();
+    expect(getAllByText("/").length).toBe(1);
+    expect(getByText("/")).toBeTruthy();
```
Comment (Jialecl) [inline lib/src/text-input/TextInput.test.js:660]: Isn't this redundant to the previous expect?

Comment (Lisa159) [reply lib/src/text-input/TextInput.test.js:660]: It is just double check, but yes, it is a bit redundant.

Comment (Lisa159) [reply lib/src/text-input/TextInput.test.js]: Done

File: lib/src/text-input/TextInput.test.js (hunk 1 above)
Comment (Lisa159) [reply lib/src/text-input/TextInput.test.js]: Done

File: lib/src/text-input/Suggestion.tsx [hunk 3]
```diff
@@ -11,7 +24,8 @@ const Suggestion = ({
   visuallyFocused,
   highlighted,
 }: SuggestionProps): JSX.Element => {
-  const regEx = new RegExp(value, "i");
+  const specialChar = transformSpecialChars(value);
```
Comment (Jialecl) [inline lib/src/text-input/Suggestion.tsx:27]: This can be in a useMemo

Comment (Lisa159) [reply lib/src/text-input/Suggestion.tsx]: Done

**Review by Jialecl — APPROVED** [review]

**Pull request merged**